app/views/my_facilities/missed_visits.html.erb

```diff
@@ -0,0 +1,130 @@
+              <%= PeriodSelection::PERIODS[:missed_visits][@selected_period] %>
+            </a>
+
+            <div class="dropdown-menu" aria-labelledby="dropdownMenuLink">
+              <% PeriodSelection::PERIODS[:missed_visits].each do |period, period_name| %>
+                <button type="submit" name="period" value=<%= period %> class="dropdown-item" form="query-filters"><%= period_name %></button>
+              <% end %>
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <table class="mt-3 mt-lg-4 table table-compact table-responsive-md table-hover" id="analytics-table">
+        <colgroup>
+          <col>
+          <% @display_periods.each do %>
+            <col class="table-divider">
+            <% unless @selected_period == :day %>
+              <col>
+            <% end %>
+          <% end %>
+          <col class="table-divider">
+          <col class="table-divider">
+          <col class="mobile">
+        </colgroup>
+        <thead>
+        <tr>
+          <th></th>
+          <% case @selected_period %>
+          <% when :quarter %>
+            <% @display_periods.reverse.each do |(year, quarter)| %>
+              <th colspan="2"><%= "Q#{quarter}-#{year}" %></th>
+            <% end %>
+          <% when :month %>
+            <% @display_periods.reverse.each do |(year, month)| %>
+              <th colspan="2"><%= month_short_name_and_year(local_month_start(year, month)) %></th>
+            <% end %>
+          <% end %>
+          <th>Calls made <sup>1</sup></th>
+          <th>ALL-TIME</th>
+        </tr>
+        <tr data-sort-method="thead" class="sorts">
+          <th class="row-label sort-label">Facilities</th>
+          <% @display_periods.reverse.each do |_| %>
+            <th class="row-label sort-label" data-sort-default>Percent</th>
+            <th class="row-label sort-label">Missed</th>
+          <% end %>
+          <th class="row-label sort-label">
+            <% case @selected_period %>
+            <% when :quarter %>
+              <%= "Q#{@display_periods.first.second}-#{@display_periods.first.first}" %>
+            <% when :month %>
+              <%= month_short_name_and_year(local_month_start(@display_periods.first.first,
+                                                              @display_periods.first.second)) %>
+            <% end %>
+          </th>
+          <th class="row-label sort-label">Registered patients <sup>2</sup></th>
+          <th class="mobile">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</th>
+        </tr>
+        </thead>
+        <tbody>
+        <tr class="row-total" data-sort-method="none">
+          <td class="type-title">All</td>
+
+          <% @display_periods.reverse.each do |period| %>
+            <td class="type-percent" data-sort-column-key="<%= period.join %>">
+              <em><%=
+                percentage(@totals_by_period.dig([period.first, period.second], :missed),
```

Comment:
Is `period` an array with two elements? Do you think a hash would make our code much more readable?

`...dig([period[:start], period[:finish])`

Reply:
Actually, I may not be understanding the data structure of `@totals_by_period`, could you help me understand?

Reply:
Yes, `period` is an array with two elements, the first element is the year, and the second is either month or quarter, all of these are integers.

totals_by_period is a hash map keyed by period, so it looks like this:
`{[2020, 2]=>{:patients=>14498, :missed=>7474}, 
[2020, 1]=>{:patients=>13222, :missed=>3868}, 
[2019, 12]=>{:patients=>11845, :missed=>4478}}`

The `period.first` and `period.second` stuff is unnecessary, I'll refactor it to just be 
`.dig(period, :missed)`.